sdk/tables/aztables/tableClient.go

```diff
@@ -7,34 +7,69 @@ import (
 	"context"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/runtime"
 )
 
 // A TableClient represents a URL to an Azure Storage blob; the blob may be a block blob, append blob, or page blob.
 type TableClient struct {
-	client *tableClient
-	cred   SharedKeyCredential
+	client  *tableClient
+	service *TableServiceClient
+	cred    SharedKeyCredential
+	name    string
 }
 
 // NewTableClient creates a TableClient object using the specified URL and request policy pipeline.
-func NewTableClient(serviceURL string, cred azcore.Credential, options *TableClientOptions) (*TableClient, error) {
-	con := newConnection(serviceURL, cred, options.getConnectionOptions())
+func NewTableClient(tableName string, serviceURL string, cred azcore.Credential, options *TableClientOptions) (*TableClient, error) {
```

Comment:
Curious, since I heard you mention that azidentity cant be used with Tables....should the credential type here be different than an azcore.Credential interface then? Because azidentity does implement that. Could it lead to some confusion?

Reply:
Good question. In other languages, the type is typically `TokenCredential` to indicate this. Based on the naming, I assumed this was not implied to be a token credential type. Is there not an equivalently named type in azcore?